tools/replicate_server_settings.rb

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env ruby
+require File.expand_path("../config/environment", __dir__)
+require 'trollop'
+
+opts = Trollop.options(ARGV) do
+  banner "USAGE:   #{__FILE__} -s <server id> -p <settings path separated by a /> \n" \
+         "Example: #{__FILE__} -d -s 1 -p ems/ems_amazon/additiona_instance_types"
+
+  opt :dry_run,  "Dry Run",                                                :short => "d"
+  opt :serverid, "Replicating source server Id (default: current server)", :short => "s"
+  opt :path,     "Replicating source path within advanced settings hash",  :short => "p", :default => ""
+end
+
+puts opts.inspect
+
+Trollop.die :path, "is required" unless opts[:path_given]
+
+server = opts[:serverid] ? MiqServer.find(opts[:serverid]) : MiqServer.my_server
+settings = server.get_config("vmdb")
+
+config = settings.config
+path = opts[:path].split("/").map(&:to_sym)
+values = config.fetch_path(path)
+target_servers = MiqServer.where.not(:id => server.id)
+
+puts "Replicating from server id=#{server.id}, path=#{opts[:path]} to #{target_servers.count} servers"
+target_servers.each do |target|
+  puts " - replicating to server id=#{target.id}..."
+  target_settings = target.get_config("vmdb")
+  target_settings.config.store_path(*path, values)
```

Comment:
Similar thing here, I think you can make use of `target.add_settings_for_resource` directly rather than going through the `set_config` method.

Reply:
Thanks! Done